web/docs/understand/formats/json.md

```diff
@@ -36,6 +36,48 @@ For example, the [Suricata EVE JSON](suricata.md) format includes a field
 `--selector=event_type:suricata`, a field value of `flow` will create a schema
 with the name `suricata.flow`.
 
+### `--unnest-separator=<string>` (Parser)
+
+Designates a string that will unnest the fields if possible.
+
+For example, the [Zeek JSON](zeek-json.md) includes a field `id` in many of
+it's events. The `id` is composed of `orig_h`, `orig_p`, `resp_h`,`resp_p`
+nested fields. The Zeek JSON outputs the `id` field as a 4 individual key-value
+pairs instead of an JSON object. This can be represented as:
+```json
+{
+  "id.orig_h" : "1.1.1.1",
+  "id.orig_p" : 10,
+  "id.resp_h" : "1.1.1.2",
+  "id.resp_p" : 5
+}
+```
+
+The `--unnest-separator` for Zeek JSON is a `.`.
+Each `.` in a field name will be treated as separator between nested fields.
+Input in the example above will be transformed into:
+```json
+{
+  "id" : {
+    "orig_h" : "1.1.1.1",
+    "orig_p" : 10,
+    "resp_h" : "1.1.1.2",
+    "resp_p" : 5
+  }
+}
+```
+
+Sometimes the separator will stay as a part of a field name. For example the:
+```json
+{
+  "cpu" : "cpu01",
+  "cpu.logger" : "logger1"
+}
+```
+
+will not be transformed. The "cpu" field has a a value so it can't be unnested
+into a JSON object.
```

Comment:
This is nice. I've got some suggestions, but it's more or less the same content:

```suggestion
### `--unnest-separator=<string>` (Parser)

A delimiter that, if present in keys, causes values to be treated as values of nested records.

A popular example of this is the [Zeek JSON](zeek-json.md) format. It includes the fields `id.orig_h`, `id.orig_p`, `id.resp_h`, and `id.resp_p` at the top-level. The data is best modeled as an `id` record with four nested fields `orig_h`, `orig_p`, `resp_h`, and `resp_p`.

Without an unnest separator, the data looks like this:

---json
{
  "id.orig_h" : "1.1.1.1",
  "id.orig_p" : 10,
  "id.resp_h" : "1.1.1.2",
  "id.resp_p" : 5
}
---

With the unnest separator set to `.`, VAST reads the events like this:

---json
{
  "id" : {
    "orig_h" : "1.1.1.1",
    "orig_p" : 10,
    "resp_h" : "1.1.1.2",
    "resp_p" : 5
  }
}
---
```

I used triple dashes over triple backticks because that otherwise breaks the suggestion mode of review comments, so please use backticks instead and also reflow the text to 80 lines.

Reply:
I unresolved this as this part is still missing:

> I used triple dashes over triple backticks because that otherwise breaks the suggestion mode of review comments, so please use backticks instead and also reflow the text to 80 lines.

Reply:
> I unresolved this as this part is still missing:
> 
> > I used triple dashes over triple backticks because that otherwise breaks the suggestion mode of review comments, so please use backticks instead and also reflow the text to 80 lines.

Sorry, i misunderstood what you wrote. Thought that i should be using the --- instead of ```. Then i thought that the please use backticks was a TYPO for dashes. And then i thought i was missing 80 character limit in the original commit

Reply:
No worries, I should've phrased that better. That was really confusing.